docs/technologies.md

```diff
@@ -0,0 +1,423 @@
+
+```python
+class BigQueryOperation(object, metaclass=ABCMeta):
+
+    @abstractmethod
+    def to_job(self, id: str, retry_count: int = DEFAULT_RETRY_COUNT, retry_pause_sec: int = DEFAULT_RETRY_PAUSE_SEC):
+        pass
+
+    def run(self, runtime=DEFAULT_RUNTIME):
+        pass
+```
+
+You can turn a lazy operation into a job or simply run it (useful for ad-hoc queries or debugging).
+
+A SQL code that you provide to the methods is templated (as mentioned in the previous section). 
+Besides a configuration parameters, you can access the
+`runtime` parameter. It's available as the `dt` variable. For example:
+
+```python
+dataset.write_truncate('target_table', '''
+SELECT *
+FROM `{another_table}`
+WHERE PARTITION_TIME = '{dt}'
+''')
+```
+
+#### Write truncate
+
+The `write_truncate` method takes a SQL query, executes it, and saves a result into a specified table.
+
+```python
+dataset.write_truncate('target_table', '''
+SELECT *
+FROM `{another_table}`
+''')
+```
+
+This method overrides the whole table or a single partition, depending on the type of a specified table.
+To specify type of a table you use, you need to use the `partitioned` parameter.
+
+```python
+dataset.write_truncate('target_table', '''
+SELECT *
+FROM `{another_table}`
+''', partitioned=False)
+```
+
+The `write_truncate` method also expects that a specified table exists. It won't create a new table from a query result.
+
+#### Write append
+
+The `write_append` method acts almost the same as the `write_truncate`. The difference is that `write_append` doesn't
+override a specified table, but appends new records.
+
+```python
+dataset.write_append('target_table', '''
+SELECT *
+FROM `{another_table}`
+''')
+```
+
+#### Write tmp
+
+The `write_tmp` method allows you to create or override a non-partitioned table from a query result.
+
+```python
+select_dataset.write_tmp('target_temporary_table', '''
+SELECT *
+FROM `{another_table}`
+''')
+```
+
+#### Collect
+
+The `collect` method allows you to fetch a query results to a [Pandas DataFrame](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html).
+
+```python
+rows: pd.Dataframe = dataset.collect('''
+SELECT *
+FROM `{another_table}`
+''').run()
+```
+
+Note that to fetch a result, you need to call the `run` method.
+
+#### Collect list
+
+The `collect_list` method works almost the same as the `collect` method,
+the difference is that it returns a Python `list`.
+
+```python
+rows: list = dataset.collect('''
+SELECT *
+FROM `{another_table}`
+''').run()
+```
+
+#### Create table
+
+The `create_table` method allows you to create a table.
+
+```python
+create_my_new_table_operation = dataset.create_table('''
+    CREATE TABLE IF NOT EXISTS my_new_table (
+      some_field STRING,
+      another_field FLOAT64)
+''')
+```
+
+#### Table sensor
+
+The `sensor` function allows you to wait for a specified table.
```

Comment:
w sensie nie Tobie, chodzi o to że workflow ma serie jobów, gdzie joby czekają na dane, to nie jest oczywiste

Reply:
done